lib/buffer.js

```diff
@@ -292,7 +292,7 @@ Buffer.concat = function(list, length) {
 
   var buffer = Buffer.allocUnsafe(length);
   var pos = 0;
-  for (let i = 0; i < list.length; i++) {
+  for (i = 0; i < list.length; i++) {
```

Comment:
So this was to address a lint issue.

Really the `var i` in the for loop before should be declared outside of the if statement, so have added another commit to fix this,